tests/integrationv2/test_session_resumption.py

```diff
@@ -102,3 +100,210 @@ def test_session_resumption_s2n_client(managed_process, cipher, curve, protocol,
         assert results.exception is None
         assert results.exit_code == 0
         assert results.stdout.count(bytes("6 server accepts that finished".encode('utf-8')))
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+def test_tls13_session_resumption_s2n_server(managed_process, tmp_path, cipher, curve, protocol, provider, certificate):
+    port = str(next(available_ports))
+
+    # Use temp directory to store session tickets
+    p = tmp_path / 'ticket.pem'
+    path_to_ticket = str(p)
+
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        insecure=True,
+        reconnect=False,
+        data_to_send=data_bytes(4069),
+        extra_flags = ['-sess_out', path_to_ticket],
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.use_session_ticket = True
+    server_options.extra_flags = None
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    # The client should have received a session ticket
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert b'Post-Handshake New Session Ticket arrived:' in results.stdout
+
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        # The first connection is a full handshake
+        assert b'Resumed session' not in results.stdout
+
+    # Client inputs received session ticket to resume a session
+    assert os.path.exists(path_to_ticket)
+    client_options.extra_flags = ['-sess_in', path_to_ticket]
+
+    port = str(next(available_ports))
+    client_options.port = port
+    server_options.port = port
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    s2n_version = get_expected_s2n_version(protocol, OpenSSL)
+
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
```

Comment:
On the OpenSSL client side, verify that handshake was a resumption handshake and not full. Does OpenSSL client print any message indicating that session was resumed using the session ticket successfully? Asserting for no exceptions and no error's aren't sufficient in this case.

Reply:
There isn't a print message that says "Session resumed" for s_client, if that's what you're looking for. Also, the resumption is already being verified on the server-side. I suppose I could add an assert that the client did not read a server certificate message? But not reading something provides less proof than reading something, imo.

Reply:
Added assert to check client did not read server cert